# Frequently Asked Questions

Для упрощения жизни и атмосферы в чатике — ответы на некоторые популярные вопросы вынесены сюда.


### Q: Локально проходят все тесты, а на сервере ошибка `flake8`/`mypy`

<details><summary><b>A: [под спойлером]</b></summary>
В первую очередь нужно проверить, что вы запускаете тесты и линтеры с учётом файла конфигурации (`setup.cfg`).  

Есть 2 варианта как запустить тесты и линтеры
* Можно запускать из корня проекта, тогда файл подцепится автоматически
  ```shell
  python -m flake8 ./path/to/the/task
  python -m mypy ./path/to/the/task
  python -m pytest ./path/to/the/task
  ```
* Можно запускать из любой директории, но нужно указать файл ручками
  ```shell
  python -m flake8 --config ../../setup.cfg task_name
  python -m mypy --config-file ../../setup.cfg task_name
  python -m pytest -c ../../setup.cfg task_name
  ```
</details>


### Q: На сервере какая-то странная ошибка про `.gitlab-ci.yml`

<details><summary><b>A: [под спойлером]</b></summary>
Перед запуском всех тестов проверяется, что файл `.gitlab-ci.yml` не был изменён.

Для этого дополнительно сравнивается файл `.gitlab-ci.yml` в вашем репозитории и самая последняя версия из публичного репозитория.
Если файлы различаются, то выкидывается ошибка.  
Возникнуть она может даже если вы не меняли файл, но в публичном репозитории он обновился.

Решение очень простое:
```shell
git pull upstream master
```
</details>
